packages/client-sdk-nodejs/src/internal/grpc/headers-interceptor.ts

```diff
@@ -36,9 +36,9 @@ export class HeaderInterceptorProvider {
     return (options, nextCall) => {
       return new InterceptingCall(nextCall(options), {
         start: (metadata, listener, next) => {
-          this.headersToAddEveryTime.forEach(h =>
-            metadata.add(h.name, h.value)
-          );
+          this.headersToAddEveryTime.forEach(h => {
+            metadata.set(h.name, h.value);
```

Comment:
Bug fix: `set` will overwrite the header value, `add` will blindly append

The auth header was continually getting appended with each retried request, thus short-circuiting the actual retried request and causing that loop of retries that didn't go anywhere

Reply:
After fixing this bug, the retries via resetting grpc deadline appear to work as expected!

Logs snippet below with bold font to highlight timing: 1 second between retries, 100ms between determining eligibility and actually retrying

[2024-08-14T21:21:21.112Z] TRACE (Momento: StorageDataClient): Issuing 'put' request; store: store, key: key
...
[2024-08-14T**21:21:22.116**Z] DEBUG (Momento: RetryInterceptor): Request eligible for retry: path: /store.Store/Put; response status code: 4; number of attempts (1); will retry in 100ms
[2024-08-14T**21:21:22.217**Z] DEBUG (Momento: StorageClientTimeoutInterceptor): intercepting call with options.deadline 1723670482113
...
[2024-08-14T**21:21:23.218**Z] DEBUG (Momento: RetryInterceptor): Request eligible for retry: path: /store.Store/Put; response status code: 4; number of attempts (2); will retry in 100ms
[2024-08-14T**21:21:23.319**Z] DEBUG (Momento: StorageClientTimeoutInterceptor): intercepting call with options.deadline 1723670483217
...
[2024-08-14T**21:21:24.320**Z] DEBUG (Momento: RetryInterceptor): Request eligible for retry: path: /store.Store/Put; response status code: 4; number of attempts (3); will retry in 100ms
[2024-08-14T**21:21:24.422**Z] DEBUG (Momento: StorageClientTimeoutInterceptor): intercepting call with options.deadline 1723670484319
...
[2024-08-14T**21:21:25.423**Z] DEBUG (Momento: RetryInterceptor): Request eligible for retry: path: /store.Store/Put; response status code: 4; number of attempts (4); will retry in 100ms
[2024-08-14T**21:21:25.523**Z] DEBUG (Momento: StorageClientTimeoutInterceptor): intercepting call with options.deadline 1723670485422
[2024-08-14T21:21:25.523Z] DEBUG (Momento: StorageClientTimeoutInterceptor): received deadline < current time, resetting
[2024-08-14T21:21:25.523Z] DEBUG (Momento: StorageClientTimeoutInterceptor): Unable to successfully retry request within client timeout, canceling request
[2024-08-14T21:21:25.523Z] DEBUG (Momento: DefaultStorageRetryStrategy): Determining whether request is eligible for retry; status code: 1, request type: /store.Store/Put, attemptNumber: 5
[2024-08-14T21:21:25.523Z] DEBUG (Momento: DefaultStorageEligibilityStrategy): Response with status code 1 is not retryable.
[2024-08-14T21:21:25.523Z] DEBUG (Momento: RetryInterceptor): Request not eligible for retry: path: /store.Store/Put; retryable status code: 1; number of attempts (5).
putInt: The request was cancelled by the server; please contact us at support@momentohq.com: 1 CANCELLED: Unable to successfully retry request within client timeout

Reply:
@anitarua awesome, thanks - do you have any more changes planned on this atm or should I pull it down and play with it some more at this point?

Reply:
no further changes planned currently -- feel free to pull it down and let me know what you think!